robyn/__init__.py

```diff
@@ -100,9 +101,25 @@ def add_route(
             route_type = http_methods[route_type]
 
         return self.router.add_route(
-            route_type, endpoint, handler, is_const, self.exception_handler
+            route_type, endpoint, handler, is_const, self.dependencies.dependency_map,self.exception_handler
         )
 
+    def inject(self, route=None, **kwargs):
+        if route:
+            self.dependencies.add_route_dependency(route, **kwargs)
+        else:
+            self.dependencies.add_global_dependency(**kwargs)
+
+    def get_injected_dependencies(self, route=None) -> dict:
+        # How will we access these from sub routes?
+        # @Darren, @Ido ?? Any ideas?
```

Comment:
Hey @IdoKendo @DarrenSeng 👋 

I have one question regarding the public facing interface of the API. Do you folks have any suggestions?